main/lsp/LSPTypechecker.cc

```diff
@@ -757,6 +757,19 @@ vector<ast::ParsedFile> LSPTypechecker::getResolved(absl::Span<const core::FileR
     return pipeline::incrementalResolve(*gs, move(updatedIndexed), move(foundHashesForFiles), config->opts, workers);
 }
 
+ast::ParsedFile LSPTypechecker::getResolved(core::FileRef fref, WorkerPool &workers) const {
+    std::vector<ast::ParsedFile> trees = this->getResolved(std::initializer_list<core::FileRef>{fref}, workers);
+
+    ENFORCE(trees.size() == 1);
+    // This is defensive. We should always get a tree back for files that have been through the slow path, but just in
+    // case we catch that here and return an empty tree.
+    if (trees.size() != 1) {
+        return ast::ParsedFile{nullptr, fref};
+    } else {
+        return std::move(trees.front());
+    }
+}
```

Comment:
I think I would have preferred to just crash here if the size was wrong. `trees.at(0)` should raise an out_of_bounds exception if we've got the wrong size, and then we'll see it in our crash logs and be able to triage more easily (vs the alternative: tracking why in some cases the `ParsedFile::tree` field is `nullptr` back to this call site).

Reply:
That's fair, I'll make a follow up to make the crash more obvious.

Reply:
#8553